### PR TITLE
feat: add setUser and getUserWithContext to client; SkylabUser builder

### PR DESCRIPTION
**Pull request by bgiori**
Add `setUser()` and `getUserWithContext()` along side a few `SkylabUser` tests.

### CODE REVIEW

**Review by curtisliu — COMMENTED** [review]
A key part of the feature request is to allow creating a new SkylabUser based on the existing SkylabUser, something like this in Android:

```java
SkylabUser user = SkylabUser.builder()
                    .copyUser(Skylab.getInstance().getUser())
                    .setUserProperty("newProperty", "value")
                    .build();
```

**Comment by bgiori** [comment]
@curtisliu Thanks for the comments. Moving forward I'll try to keep better parity with android during this beta phase. I updated the PR to include the `SkylabUser.Builder` like in android in order to provide the feature that was requested. It should look pretty familiar :)

File: Sources/Skylab/SkylabUser.swift [hunk 1]
```diff
@@ -7,61 +7,182 @@
...
+                self.user.deviceModel = deviceModel
+            }
+            if let carrier = user.carrier {
+                self.user.carrier = carrier
+            }
+            if let library = user.library {
+                self.user.library = library
+            }
+            if let userProperties = user.userProperties {
+                self.user.userProperties = userProperties
```
Comment (curtisliu) [inline Sources/Skylab/SkylabUser.swift:178]: Is this safe to do? Would this share userProperties object between the copies and produce the same bug I had introduced in Android? (Not sure how doing a copy of the userProperties object would affect Equatable though)

**Review by curtisliu — APPROVED** [review]

**Review by curtisliu — APPROVED** [review]

File: Sources/Skylab/SkylabUser.swift (hunk 1 above)
Comment (bgiori) [reply Sources/Skylab/SkylabUser.swift:178]: Good question. Quick test:

```swift
var map: [String:String] = [:]
var map2 = map
map2["test"] = "test"
print(map)
print(map2)
```

output
```
[:]
["test": "test"]
```

so we should be in the clear

**Pull request merged**